lib/connectors/mongodb/connector.rb

```diff
@@ -50,6 +51,23 @@ def self.configurable_fields
         }
       end
 
+      def self.validate_filtering(filtering = {})
+        valid_filtering = { :state => Core::Filtering::ValidationStatus::VALID, :errors => [] }
+
+        return valid_filtering unless filtering.present?
+
+        advanced_filter_config = filtering[:advanced_snippet] || {}
+        filter_keys = advanced_filter_config&.keys
+
+        if !filter_keys&.empty? && (filter_keys.size != 1 || !ALLOWED_TOP_LEVEL_FILTER_KEYS.include?(filter_keys[0]&.to_s))
```

Comment:
I would like to introduce MongoDB filtering validation as a separate PR as this one is already pretty big. 

I did some research on query validation and for now it doesn't seem to be the case, that mongo clients expose this functionality. Afaik every client lets the database handle this validation (acting as a single point of truth/validation for queries). I've added the idea of generating a query parser (which would act as a validator in our case) for MongoDB with Bison, because the grammar file for the MongoDB query language is open source.